### PR TITLE
AJAX request in Django (Using Axios)

### PR DESCRIPTION
**Pull request by atoyegbe**
### Article Checklist
**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process, quality, and style-once published, you can then submit any number of articles.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by hectorkambow** [comment]
@geoffrey45  will be reviewing this PR 👍

**Comment by hectorkambow** [comment]
plagiarism check done @geoffrey45

File: articles/ajax-request-in-django-using-axios/index.md [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+The above code 
+   - #a - Adding a submit event listener to the form.
+   - #b - Creating a new Form in JavaScript.
+   - #c - adding a CRSF token, if we do not include this we get a 403 forbidden response we won’t be able to submit the form data.
+   - #d - Here we use **axios.post** method to submit form data.
+   - #e - A alert message **Form Submitted** pop on your windows if the form submitted successfully.
+   - #f - This catches and displays the error in the console if an error occurs when submitting the form.
+
+Your **home.html** file should exactly like this.
+
```
Comment (geoffrey45) [inline articles/ajax-request-in-django-using-axios/index.md:140]: There are some missing words. Edit removing 'exactly' to look like this: 'Your `home.html` should look like this.'

**Review by geoffrey45 — CHANGES_REQUESTED** [review]
Great article @atoyegbe. Just a few minor issues. Please fix them and ping me when you are done.

**Comment by atoyegbe** [comment]
@geoffrey45 I'm done.

File: articles/ajax-request-in-django-using-axios/index.md [hunk 2]
```diff
@@ -0,0 +1,246 @@
+### What is AJAX? 
+
+Ajax is an acronym for Asynchronous JavaScript and XML. It’s not a language, framework or a web library. It’s a web technology that sends and receives data from a client to a server asynchronously all done in the background behind the scenes without needing to reload the current web page. Even though XML is in the acronym, it’s rarely used in AJAX anymore, it has been replaced by JSON (JavaScript Object Notation). JSON is easy to use, faster and much more popular compared to XML.
+
+### Prerequisite
+
+1. [Python](https://www.python.org/downloads/) and above installed on your computer.
+
```
[the remark this answers is not in the document]
Comment (geoffrey45) [reply articles/ajax-request-in-django-using-axios/index.md]: @atoyegbe Replace the words 'and above' with 'or newer'.

File: articles/ajax-request-in-django-using-axios/index.md [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+2. Python package manager, [pip](https://pypi.org/project/pip/) installed on your computer.
+
+3. knowledge of [Django]()
+
```
[the remark this answers is not in the document]
Comment (geoffrey45) [reply articles/ajax-request-in-django-using-axios/index.md]: Start these list sentences with a capital letter

File: articles/ajax-request-in-django-using-axios/index.md [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+Let’s make sure nothing is broken in our application, change into the project directory and run the command below.
+
+```bash 
+python manage.py runserver
+```
+After running the above command you will see the below image in your browser.
+![form page](/engineering-education/ajax-request-in-django-using-axios/form_page.png)
+
+Let’s go into the **templates/home.html** and we will add to our project the **Axios** library by adding the CDN script tag below outside the div tag.
+
```
[the remark this answers is not in the document]
Comment (geoffrey45) [reply articles/ajax-request-in-django-using-axios/index.md]: @atoyegbe  Change this to read: '... the CDN script tag below `</div>` closing tag'

**Comment by atoyegbe** [comment]
@geoffrey45 done.

File: articles/ajax-request-in-django-using-axios/index.md [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+         .then(res => alert(“Form Submiited”))
+         .catch(errors => console.log(errors))
+    })
+</script>
+{% endblock %}
+```
+
+### Handling Post Request in Django views.
+Navigate to the note directory and add the code below into views.py
+
```
Comment (geoffrey45) [inline articles/ajax-request-in-django-using-axios/index.md:175]: Put file directories into code blocks .eg. \`views.py`

File: articles/ajax-request-in-django-using-axios/index.md [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+<script src="https://cdnjs.cloudflare.com/ajax/libs/axios/0.21.1/axios.min.js"></script>
+
+{% endblock %}
+
+```
+
+### Using Axios to submit a form.
+>Axios is a Javascript library used to make HTTP requests from node.js or XMLHttpRequests from the browser. It is a lightweight HTTP client based on the XMLHttpRequests service. It is similar to the Fetch API and is used to perform HTTP requests.
+
```
Comment (geoffrey45) [inline articles/ajax-request-in-django-using-axios/index.md:97]: Move this explanation somewhere before the code work. Create a sub-heading with the name "What is Axios" and put the explanation in it.  Also, provide relevant links for Fetch API and XMLHttpRequests.

**Review by geoffrey45 — CHANGES_REQUESTED** [review]
@atoyegbe Make these changes and ping me when done.

**Comment by atoyegbe** [comment]
@geoffrey45 done.

File: articles/ajax-request-in-django-using-axios/index.md [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+ ```
+
+Run the command below to install the dependencies needed for this project to work.
+
+```bash 
+$ pip install -r requirements.txt
+```
+
+Let’s make sure nothing is broken in our application, change into the project directory and run the command below.
+
```
[the remark this answers is not in the document]
Comment (geoffrey45) [reply articles/ajax-request-in-django-using-axios/index.md]: @atoyegbe Break down this into two smaller sentences. Also, edit to look like this: '... directory by running `cd notes` and run the command below.'

**Review by geoffrey45 — APPROVED** [review]

**Comment by geoffrey45** [comment]
@hectorkambow tech review done. Ready for final review.

**Pull request merged**